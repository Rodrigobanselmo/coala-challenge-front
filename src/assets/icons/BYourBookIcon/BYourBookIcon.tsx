import React, { FC } from "react";

import { IIconMuiProps } from "@/core/interfaces/IIconMuiProps";
import LocalLibraryIcon from "@mui/icons-material/LocalLibrary";

export const BYourBookIcon: FC<IIconMuiProps> = ({ ...props }) => {
  return <LocalLibraryIcon {...props} />;
};

export default LocalLibraryIcon;
