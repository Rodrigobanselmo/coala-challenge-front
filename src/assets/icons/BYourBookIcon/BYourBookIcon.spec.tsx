import React from "react";
import { render, screen } from "@testing-library/react";
import { BYourBookIcon } from "./BYourBookIcon";

describe("BYourBookIcon", () => {
  it("renders without errors", () => {
    render(<BYourBookIcon />);
    const icon = screen.getByTestId("LocalLibraryIcon");

    expect(icon).toBeInTheDocument();
    expect(icon).toMatchSnapshot();
  });

  it("applies custom props", () => {
    render(<BYourBookIcon fontSize="large" color="primary" />);

    const icon = screen.getByTestId("LocalLibraryIcon");
    expect(icon).toMatchSnapshot();
  });
});
