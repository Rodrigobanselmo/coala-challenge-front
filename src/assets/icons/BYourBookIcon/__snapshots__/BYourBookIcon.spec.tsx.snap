// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BYourBookIcon applies custom props 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
  data-testid="LocalLibraryIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M12 11.55C9.64 9.35 6.48 8 3 8v11c3.48 0 6.64 1.35 9 3.55 2.36-2.19 5.52-3.55 9-3.55V8c-3.48 0-6.64 1.35-9 3.55zM12 8c1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3 1.34 3 3 3z"
  />
</svg>
`;

exports[`BYourBookIcon renders without errors 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="LocalLibraryIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M12 11.55C9.64 9.35 6.48 8 3 8v11c3.48 0 6.64 1.35 9 3.55 2.36-2.19 5.52-3.55 9-3.55V8c-3.48 0-6.64 1.35-9 3.55zM12 8c1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3 1.34 3 3 3z"
  />
</svg>
`;
